### PR TITLE
Update Android build to use Gradle 8.1 and JDK 1.17

### PR DESCRIPTION
**Pull request by MarkY-LunarG**

### CODE REVIEW

**Comment by MarkY-LunarG** [comment]
This is dependent upon CI supporting [work being done here](https://github.com/LunarG/VulkanTests/pull/602)

**Review by ziga-lunarg — APPROVED** [review]

File: .github/workflows/sdk_android_build.yml [hunk 1]
```diff
@@ -38,6 +38,10 @@ jobs:
       uses: actions/checkout@v4
       with:
         submodules: 'recursive'
+    - uses: actions/setup-java@v4
+      with:
+      distribution: 'oracle'
+      java-version: '17'
```
Comment (lunarpapillo) [inline .github/workflows/sdk_android_build.yml:44]: Indentation here looks incorrect.  Are you mixing spaces and tabs?

**Review by lunarpapillo — CHANGES_REQUESTED** [review]
IMHO, you'll also need to modify `ci_build.yml` to get the **GitHub Actions** CI working.

But no, I'm not an official developer/reviewer, so you can ignore me. :stuck_out_tongue:

File: .github/workflows/ci_build.yml [hunk 2]
```diff
@@ -211,6 +211,11 @@ jobs:
     - name: Update submodules
       run: |
         git submodule update --init
+    - name: Set up Java
+      uses: actions/setup-java@v2
+      with:
+        java-version: 17
+        distribution: "temurin"
```
Comment (lunarpapillo) [inline .github/workflows/ci_build.yml:218]: This looks good to me (although I don't know what `temurin` is).

The same lines will be needed in `sdk_android_build.yaml`, right?  Or does GFXR not use that any more?

File: .github/workflows/sdk_android_build.yml (hunk 1 above)
Comment (MarkY-LunarG) [reply .github/workflows/sdk_android_build.yml]: I think it's correct now.

File: .github/workflows/ci_build.yml (hunk 2 above)
Comment (MarkY-LunarG) [reply .github/workflows/ci_build.yml:218]: I misunderstood and thought you eant that it was only needed in ci_build.yaml.  It is now in both.  I tried `oracle` for the JDK and it didn't work (even though that used to).  Do you have an alternative?

**Comment by MarkY-LunarG** [comment]
@lunarpapillo is the Android failure something you've seen before?  The build appears to execute but it fails when attempting to find build artifacts.

**Comment by MarkY-LunarG** [comment]
Still need to have help from @lunarpapillo  figuring out why Github fails Android.  All changes that I understand are required have been made.  So I must be missing something, but I don't know how to get the entire build log either.

To me, the [logs](https://productionresultssa16.blob.core.windows.net/actions-results/76b4d28d-b800-48f8-918f-789612412546/workflow-job-run-bdbd57b1-5a74-54af-bd7f-6070ffb23d68/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-08-26T18%3A38%3A06Z&sig=G2nMlOAKdHFtN4MFr0X1ol1PKrUuiY4xLbb0sh%2FI1Ho%3D&ske=2024-08-27T04%3A29%3A38Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2024-08-26T16%3A29%3A38Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2024-05-04&sp=r&spr=https&sr=b&st=2024-08-26T18%3A28%3A01Z&sv=2024-05-04) look like there are some extra characters getting added to some of the paths:

```bash
2024-08-23T21:58:12.3842031Z BUILD SUCCESSFUL in 24m 33s
2024-08-23T21:58:12.3842853Z 165 actionable tasks: 165 executed
2024-08-23T21:58:12.7382809Z ##[group]Run mkdir gfxreconstruct-dev
2024-08-23T21:58:12.7383316Z [36;1mmkdir gfxreconstruct-dev[0m
2024-08-23T21:58:12.7383757Z [36;1mmkdir gfxreconstruct-dev/layer[0m
2024-08-23T21:58:12.7384317Z [36;1mmkdir gfxreconstruct-dev/tools[0m
2024-08-23T21:58:12.7384745Z [36;1mcp LICENSE.txt gfxreconstruct-dev/[0m
2024-08-23T21:58:12.7385285Z [36;1mcp LICENSE_ThirdParty.txt gfxreconstruct-dev/[0m
2024-08-23T21:58:12.7385901Z [36;1mcp USAGE_android.md gfxreconstruct-dev/[0m
2024-08-23T21:58:12.7386460Z [36;1mcp layer/vk_layer_settings.txt gfxreconstruct-dev/[0m
2024-08-23T21:58:12.7387261Z [36;1mcp -r android/layer/build/intermediates/cmake/release/obj/arm64-v8a gfxreconstruct-dev/layer/[0m
2024-08-23T21:58:12.7388370Z [36;1mcp -r android/layer/build/intermediates/cmake/release/obj/armeabi-v7a gfxreconstruct-dev/layer/[0m
2024-08-23T21:58:12.7389445Z [36;1mcp android/tools/replay/build/outputs/apk/debug/replay-debug.apk gfxreconstruct-dev/tools/[0m
2024-08-23T21:58:12.7390442Z [36;1mcp android/scripts/gfxrecon.py gfxreconstruct-dev/tools/[0m
2024-08-23T21:58:12.7484403Z shell: /usr/bin/bash -e {0}
2024-08-23T21:58:12.7484920Z env:
2024-08-23T21:58:12.7485383Z   JAVA_HOME: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/17.0.12-7/x64
2024-08-23T21:58:12.7486108Z   JAVA_HOME_17_X64: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/17.0.12-7/x64
2024-08-23T21:58:12.7486734Z ##[endgroup]
2024-08-23T21:58:12.7704398Z cp: cannot stat 'android/layer/build/intermediates/cmake/release/obj/arm64-v8a': No such file or directory

```

But I could be wrong.

**Comment by lunarpapillo** [comment]
> To me, the logs look like there are some extra characters getting added to some of the paths:

I think Brad is right - those look like terminal color sequences.

> Still need to have help from @lunarpapillo figuring out why Github fails Android.

I see the issue is closed - is help still needed, or is there another issue?

**Comment by MarkY-LunarG** [comment]
I didn't intentionally close it.  Sorry.  Yes this is still an issue.  The change is already merged in the `openxr-experimental` branch and so fixing this for Github is important.

**Comment by lunarpapillo** [comment]
> I didn't intentionally close it. Sorry. Yes this is still an issue. The change is already merged in the `openxr-experimental` branch and so fixing this for Github is important.

I have a few cycles to look.  I'll be doing some experiments.

**Comment by bradgrantham-lunarg** [comment]
From the logs, I see this:
```
2024-08-26T21:13:17.3091642Z "Install Android SDK Platform-Tools v.35.0.2" finished.
2024-08-26T21:13:18.2088128Z
2024-08-26T21:13:18.2089150Z > Task :layer:preBuild UP-TO-DATE
2024-08-26T21:13:18.2089910Z > Task :layer:preReleaseBuild UP-TO-DATE
2024-08-26T21:13:20.2085935Z
2024-08-26T21:13:20.2087235Z > Task :layer:configureCMakeRelWithDebInfo[arm64-v8a]
```
It looks to me like with the upgrade to new Gradle and JDK that the objects are in a new place and "Release" builds "RelWithDebInfo" (not sure that's what we want), and so the assembly of artifacts has to change to suit.  I'm guessing you want the objects from:
```
layer/build/intermediates/cxx/RelWithDebInfo/$THINGY/obj/arm64-v8a/libVkLayer_gfxreconstruct.so
```
where "THINGY" on my machine is "11y5h286" but I would guess it changes from build to build, some kind of hash.
Same is true for the Debug objects.  Probably this isn't the end of this but maybe this can help you get over an obstacle?

**Comment by mikes-lunarg** [comment]
Do the paths in ci_build.yml need to be updated?
```
    - name: Prepare artifacts
      run: |
        mkdir gfxreconstruct-dev
        mkdir gfxreconstruct-dev/layer
        mkdir gfxreconstruct-dev/tools
        cp LICENSE.txt gfxreconstruct-dev/
        cp LICENSE_ThirdParty.txt gfxreconstruct-dev/
        cp USAGE_android.md gfxreconstruct-dev/
        cp layer/vk_layer_settings.txt gfxreconstruct-dev/
        cp -r android/layer/build/intermediates/cmake/release/obj/arm64-v8a gfxreconstruct-dev/layer/
        cp -r android/layer/build/intermediates/cmake/release/obj/armeabi-v7a gfxreconstruct-dev/layer/
        cp android/tools/replay/build/outputs/apk/debug/replay-debug.apk gfxreconstruct-dev/tools/
        cp android/scripts/gfxrecon.py gfxreconstruct-dev/tools/
```
When I build this branch locally, I get outputs in a different directory than it is expecting
```
 find android | grep libVkLayer_gfxreconstruct.so$
android/layer/build/intermediates/cxx/RelWithDebInfo/11y5h286/obj/arm64-v8a/libVkLayer_gfxreconstruct.so
android/layer/build/intermediates/cxx/RelWithDebInfo/11y5h286/obj/armeabi-v7a/libVkLayer_gfxreconstruct.so
android/layer/build/intermediates/cxx/RelWithDebInfo/11y5h286/obj/x86/libVkLayer_gfxreconstruct.so
android/layer/build/intermediates/cxx/RelWithDebInfo/11y5h286/obj/x86_64/libVkLayer_gfxreconstruct.so
android/layer/build/intermediates/merged_native_libs/release/out/lib/armeabi-v7a/libVkLayer_gfxreconstruct.so
android/layer/build/intermediates/merged_native_libs/release/out/lib/arm64-v8a/libVkLayer_gfxreconstruct.so
android/layer/build/intermediates/merged_native_libs/release/out/lib/x86/libVkLayer_gfxreconstruct.so
android/layer/build/intermediates/merged_native_libs/release/out/lib/x86_64/libVkLayer_gfxreconstruct.so
android/layer/build/intermediates/stripped_native_libs/release/out/lib/x86/libVkLayer_gfxreconstruct.so
android/layer/build/intermediates/stripped_native_libs/release/out/lib/arm64-v8a/libVkLayer_gfxreconstruct.so
android/layer/build/intermediates/stripped_native_libs/release/out/lib/armeabi-v7a/libVkLayer_gfxreconstruct.so
android/layer/build/intermediates/stripped_native_libs/release/out/lib/x86_64/libVkLayer_gfxreconstruct.so
android/layer/build/intermediates/library_and_local_jars_jni/release/jni/x86/libVkLayer_gfxreconstruct.so
android/layer/build/intermediates/library_and_local_jars_jni/release/jni/arm64-v8a/libVkLayer_gfxreconstruct.so
android/layer/build/intermediates/library_and_local_jars_jni/release/jni/armeabi-v7a/libVkLayer_gfxreconstruct.so
android/layer/build/intermediates/library_and_local_jars_jni/release/jni/x86_64/libVkLayer_gfxreconstruct.so
android/layer/build/intermediates/library_jni/release/jni/armeabi-v7a/libVkLayer_gfxreconstruct.so
android/layer/build/intermediates/library_jni/release/jni/arm64-v8a/libVkLayer_gfxreconstruct.so
android/layer/build/intermediates/library_jni/release/jni/x86/libVkLayer_gfxreconstruct.so
android/layer/build/intermediates/library_jni/release/jni/x86_64/libVkLayer_gfxreconstruct.so
android/layer/build/.transforms/e7d6d5b36a275566b815ab100bfe03db/transformed/out/jni/arm64-v8a/libVkLayer_gfxreconstruct.so
android/layer/build/.transforms/e7d6d5b36a275566b815ab100bfe03db/transformed/out/jni/armeabi-v7a/libVkLayer_gfxreconstruct.so
android/layer/build/.transforms/e7d6d5b36a275566b815ab100bfe03db/transformed/out/jni/x86/libVkLayer_gfxreconstruct.so
android/layer/build/.transforms/e7d6d5b36a275566b815ab100bfe03db/transformed/out/jni/x86_64/libVkLayer_gfxreconstruct.so
android/tools/replay/build/intermediates/merged_native_libs/release/out/lib/armeabi-v7a/libVkLayer_gfxreconstruct.so
android/tools/replay/build/intermediates/merged_native_libs/release/out/lib/arm64-v8a/libVkLayer_gfxreconstruct.so
android/tools/replay/build/intermediates/merged_native_libs/release/out/lib/x86/libVkLayer_gfxreconstruct.so
android/tools/replay/build/intermediates/merged_native_libs/release/out/lib/x86_64/libVkLayer_gfxreconstruct.so
android/tools/replay/build/intermediates/stripped_native_libs/release/out/lib/arm64-v8a/libVkLayer_gfxreconstruct.so
android/tools/replay/build/intermediates/stripped_native_libs/release/out/lib/armeabi-v7a/libVkLayer_gfxreconstruct.so
android/tools/replay/build/intermediates/stripped_native_libs/release/out/lib/x86/libVkLayer_gfxreconstruct.so
android/tools/replay/build/intermediates/stripped_native_libs/release/out/lib/x86_64/libVkLayer_gfxreconstruct.so
```
Note that there is no `android/layer/build/intermediates/cmake/`

**Comment by bradgrantham-lunarg** [comment]
I've attempted a fix at https://github.com/LunarG/gfxreconstruct/pull/1683 but have made it against `dev` so I could get a GH CI run.  I think probably easy to just apply that change to marky_update_gradle_to_8.1

**Comment by bradgrantham-lunarg** [comment]
> I've attempted a fix at #1683 but have made it against `dev` so I could get a GH CI run. I think probably easy to just apply that change to marky_update_gradle_to_8.1

https://github.com/MarkY-LunarG/gfxreconstruct/pull/9

**Comment by MarkY-LunarG** [comment]
I've pushed this change up into our `openxr-experimental` branch where it was most needed for now.  But I would still like approval here before I push it up.  @lunarpapillo , does everything look good?

**Comment by bradgrantham-lunarg** [comment]
I'm not sure this affects our internal CI, so we can consider this approved or not inside the GFXR team.

**Pull request merged**